src/cdk/text-field/autosize.ts

```diff
@@ -203,6 +211,13 @@ export class CdkTextareaAutosize implements AfterViewInit, DoCheck, OnDestroy {
     textarea.placeholder = placeholderText;
 
     this._previousValue = value;
+
+    if ((textarea.scrollHeight !== this._previousHeight && !this._maxHeight)
+        || (textarea.scrollHeight !== this._previousHeight
```

Comment:
I think this logic might miss one case. Let's say we have:
```
_previousHeight = 50
_maxHeight = 100
textarea.scrollHeight = 200
```
This will not trigger an event, even though it should. It also doesn't take the minHeight into account at all. I would do this:

```ts
const scrollHeight = textarea.scrollHeight;
const newHeight = Math.max(
    Math.min(scrollHeight, this._maxHeight || Infinity),
    this._minHeight || -Infinity);
if (newHeight != this._previousHeight) {
  ...
}
```